lib/openhab/core/items/semantics.rb

```diff
@@ -390,6 +388,107 @@ def points(*point_or_property_types)
+          #
+          #   @param [kwargs] **tags Exactly one pair of `tag` => `parent` where tag is either a Symbol or a String
+          #     for the tag to be added, and parent is either a {Tag}, a symbol or a string of an existing tag.
+          #   @return [Array<Tag>] An array of tags successfully added.
+          #
+          #   @example Add one semantic tag `Balcony` whose parent is `Semantics::Outdoor` (Location)
+          #     Semantics.add(Balcony: Semantics::Outdoor)
+          #
+          #   @example Add multiple semantic tags
+          #     Semantics.add(Balcony: Semantics::Outdoor,
+          #                   SecretRoom: Semantics::Room,
+          #                   Motion: Semantics::Property)
+          #
+          # @overload self.add(label: nil, synonyms: "", description: "", **tags)
+          #   Add a custom semantic tag with extra details.
+          #
+          #   @example
+          #     Semantics.add(SecretRoom: Semantics::Room, label: "My Secret Room",
+          #       synonyms: "HidingPlace", description: "A room that requires a special trick to enter")
+          #
+          #   @param [String,nil] label Optional label. When nil, infer the label from the tag name,
+          #     converting `CamelCase` to `Camel Case`
+          #   @param [String,Array<String,Symbol>] synonyms An array of synonyms, or a string containing a
+          #     comma separated list of synonyms for this tag.
+          #   @param [String] description A longer description of the tag.
+          #   @param [kwargs] **tags Exactly one pair of `tag` => `parent` where tag is either a Symbol or a String
+          #     for the tag to be added, and parent is either a {Tag}, a symbol or a string of an existing tag.
+          #   @return [Array<Tag>] An array of tags successfully added.
+          #
+          def self.add(label: nil, synonyms: "", description: "", **tags)
+            raise "Tags must be specified" if tags.empty?
+            if (tags.length > 1) && !(label.nil? && synonyms.empty? && description.empty?)
+              raise "Additional options can only be specified when creating one tag"
+            end
+
+            synonyms = synonyms.map(&:to_s).map(&:strip).join(",") if synonyms.is_a?(Array)
+
+            tags.map do |name, parent|
+              parent_is_tag = parent.respond_to?(:java_class) && parent.java_class < Tag.java_class
+              parent = parent_is_tag ? parent.java_class : parent.to_s
+              name = name.to_s
+              org.openhab.core.semantics.SemanticTags.add(name, parent, label, synonyms, description)
+                                                    &.then { const_missing(name) }
+            end.compact
+          end
+        end
+
+        #
+        # Returns all available Semantic tags
+        #
+        # @return [Enumerator] an enumerator containing all the Semantic tags
```

Comment:
This returns `Array<Module>`, not `Enumerator`

Reply:
Thanks for picking that up